content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,809 @@
+   the execution will be canceled and an error returned.
+   Use `0` for no timeout.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field contains the message ID.
+
+### `extern char* waku_relay_enough_peers(int nodeId, char* pubsubTopic)`
+
+Determine if there are enough peers to publish a message on a given pubsub topic.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-jsonconfig).
+2. `char* pubsubTopic`: Pubsub topic to verify.
+   If `NULL`, it verifies the number of peers in the default pubsub topic.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field contains a `boolean` indicating whether there are enough peers.
+
+For example:
+
+```json
+{
+  "result": true
+}
+```
+
+### `extern char* waku_relay_subscribe(int nodeId, char* pubsubTopic)`
+
+Subscribe to a Waku Relay pubsub topic to receive messages.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-jsonconfig).
+2. `char* pubsubTopic`: Pubsub topic to subscribe to. 
+   If `NULL`, it subscribes to the default pubsub topic.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field contains an `integer` that is the subscription id.
+
+For example:
+
+```json
+{
+  "result": 0
+}
+```
+
+**Events**
+
+When a message is received, a ``"message"` event` is emitted containing the message, pubsub topic, and node ID in which
+the message was received.
+
+The `event` type is [`JsonMessageEvent`](#jsonmessageevent-type).
+
+For Example:
+
+```json
+{
+  "nodeId": 1,
+  "type": "message",
+  "event": {
+    "subscriptionID": 1,
+    "pubsubTopic": "/waku/2/default-waku/proto",
+    "messageID": "0x6496491e40dbe0b6c3a2198c2426b16301688a2daebc4f57ad7706115eac3ad1",
+    "wakuMessage": {
+      "payload": "TODO",
+      "contentTopic": "/my-app/1/notification/proto",
+      "version": 1,
+      "timestamp": 1647826358000000000
+    }
+  }
+}
+```
+
+### `extern char* waku_relay_close_subscription(int nodeId, char* subscriptionId)`
+
+Closes a Waku Relay subscription.
+No more messages will be received from this subscription.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-jsonconfig).
+2. `char* subscriptionId`: Subscription ID to close.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse-type).
+If the execution is successful, the `result` field is set to `null`.
+
+For example:
+
+```json
+{
+   "result": null
+}
+```
+
+### `extern char* waku_relay_unsubscribe_from_topic(int nodeId, char* topic)`
+
+Closes the pubsub subscription to a pubsub topic.
+Existing subscriptions will not be closed, but they will stop receiving messages.
```

Comment:
Not sure I am getting that. What is the difference with `waku_relay_close_subscription` ? Why are existing not being closed yet it closes a subscription?

Reply:
This is because on go-waku you can have different subscriptions to the same topic. There's this concept of broadcaster/multiplexer that takes all the messages received in a pubsub topic, and shares it to all the subscribers interested in receiving messages from this topic. I ended up simplifying this and only having a single subscription.

Reply:
Does this mean this API goes away after the simplification?

Reply:
Yes. I removed it already from go-waku c-bindings